src/driver/kvm.rs

```diff
@@ -162,6 +165,41 @@ impl<T: KVMIntrospectable> Introspectable for Kvm<T> {
         Ok(())
     }
 
+    fn get_page_access(&self, paddr: u64) -> Result<u8, Box<dyn Error>> {
+        Ok(self.kvmi.get_page_access(paddr).unwrap().try_into()?)
+    }
+
+    fn set_page_access(&self, paddr: u64, access: u8) -> Result<(), Box<dyn Error>> {
```

Comment:
I would prefer an enum for `access`

Reply:
You are just passing the access permissions without any abstraction here. This is a bad idea because hypervisors have differing conventions for access permissions. Now, I understand that you want to be able to do arithmetics with the values so a plain enum would problably not be the best fit. Instead I suggest you have a look at the `bitflags` crate.